crates/ide_assists/src/handlers/remove_unused_param.rs

```diff
@@ -53,12 +53,11 @@ pub(crate) fn remove_unused_param(acc: &mut Assists, ctx: &AssistContext) -> Opt
     }
 
     let mut param_position = func.param_list()?.params().position(|it| it == param)?;
-    // param_list() does not take self param into consideration, hence this additional check is
-    // added. There are two cases to handle in this scenario, where functions are
-    // associative(functions not associative and not containting contain self, are not allowed), in
-    // this case param position is rightly set. If a method call is present which has self param,
-    // that needs to be handled and is added below in process_usage function to reduce this increment and
-    // not consider self param.
+    // param_list() does not take the self param into consideration, hence this additional check
```

Comment:
r? @feniljain the original phrasing here seemed a bit unfortunate, can you take a look?

Reply:
Yeah, on second read, it does seem confusing, thank you for the fix, also for this phrase, ```(as inherent functions not containting self are not allowed)```. I actually meant inherent function **calls** with self attribute are not allowed, for eg. 
```rust
S.foo(&S, 9); //Error
S.foo(9); //Correct
impl S {
    fn foo(&self, _a: i32) {}
}
```

And on second thought instead of keeping this over here, we can shift it over to where we are talking about inherent functions, making the whole comment as:

```
param_list() does not take the self param into consideration, hence this additional check is required. There are two cases to handle in this scenario, first being of them associated functions, in which case param_position is correctly incremented. The second case is of inherent functions, which is handled below in process_usage by reverting the increment made here(as such inherent function containing self reference are not allowed).
```

Reply:
Updated, can you take a look? It still doesn't sound great, but I suppose it's better.